doc/manual/manual.tex

```diff
@@ -5,183 +5,241 @@
-  \item{\textbf{FIFO interface.} Communicates with ALE through a text interface (Section \ref{sec:pipes_interface}).}
-  \item{\textbf{RL-Glue interface.} Communicates with ALE via RL-Glue (Section \ref{sec:rlglue_interface}).}
-\end{itemize}
+We also discuss some sensible features of this environment, such as its stochasticity.
 
-\section{Installing}
+\section{Installing}\label{sec:install}
 
 \subsection{Requirements}
 
-To build and run ALE, you will need:
+The basic requirements to build and run the ALE are:
 
 \begin{itemize}
-  \item{g++ / make}
+  \item \verb+CMake / make / g+++
+\end{itemize}
+Notice that in the past (previous to version 0.5.0) the recommended method to compile the ALE was
+through Makefiles written by the ALE developers. This has changed and now we suggest everyone to
+use CMake to generate a Makefile and then to compile the code using such generated Makefile. The
+old Makefiles written for Linux and OS X (files \verb+makefile.mac+ and \verb+makefile.unix+) are
+still available, but we do not support them anymore.
+
+The ALE provides some functionalities that are not activated by default, but if chosen the user is 
+required to use additional packages:
+\begin{itemize}
+  \item \verb+SDL / RL-Glue+
 \end{itemize}
 
 \subsection{Installation/Compilation}
 
-This tutorial assumes that you have extracted ALE to \verb+ale_0_4+. Compiling
-ALE on a UNIX machine is as simple as:
+One first has to install the ALE requirements before compiling the ALE itself. Assuming that 
+\verb+g+++ and \verb+make+ are already available to the user, to install CMake and SDL is 
+straightforward, and it can be done through package managers in both OS X and Linux:
 
+\subsubsection*{OS X:}
 \begin{verbatim}
-> cd ale_0_4
-ale_0_4> cp makefile.unix makefile
-ale_0_4> make
+  > brew install cmake
+  > brew install sdl
 \end{verbatim}
 
-Alternate makefiles are provided for 
+\subsubsection*{Linux (e.g.: Ubuntu):}
+\begin{verbatim}
+  > sudo apt-get install cmake
+  > sudo apt-get install libsdl1.2-dev
+\end{verbatim}
 
-\begin{itemize}
-  \item{Mac OS X: \verb+makefile.mac+}
-\end{itemize}
+To install RL-Glue, in both systems, we recommend the user to go to the RL-Glue 
+webpage\footnote{\url{http://glue.rl-community.org/wiki/Main_Page}} and to follow
+the presented instructions.
 
-\subsection{Sample Agents}
+We are going to assume the ALE was extracted to \verb+ale_0_5+. Then, compiling it
+with CMake is very simple (in both systems):
+\begin{verbatim}
+  > cd ale_0_5
+  ale_0_5> cmake .
+  ale_0_5> make -j4
+\end{verbatim}
 
-The following sample agents are available:
+This compiles the code without SDL and RL-Glue. If one wants to compile ALE with such libraries it 
+is enough to change such flags in the file \verb+ale_0_5/CMakeLists.txt+. More specifically, lines
+3 and/or 4 have to be changed (by replacing the word \verb+OFF+ by \verb+ON+ in the
+correspondent line):
```

Comment:
We may also want to present the other method of installing:

cmake -DUSE_SDL=ON/OFF -DUSE_RLGLUE=ON/OFF .

This avoids having to edit CMakeLists.txt